deployment/lightgbm/lightgbm-house-price-prediction.ipynb

```diff
@@ -0,0 +1,488 @@
+      "metadata": {
+        "id": "Gj8pVhlFKBRr"
+      },
+      "outputs": [],
+      "source": [
+        "y_pred = gbm.predict(X_test)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "qoI0lnidLF9Z"
+      },
+      "outputs": [],
+      "source": [
+        "MAE = mean_absolute_error(y_test, y_pred)\n",
+        "MSE = mean_squared_error(y_test, y_pred)\n",
+        "RMSE = mean_squared_error(y_test, y_pred, squared=False) "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "I56DHbNWKO6f",
+        "outputId": "cbe4b6c0-41a5-499d-88b5-bd2d380bbe5f"
+      },
+      "outputs": [],
+      "source": [
+        "print(f\"MAE: {round(MAE, 3)}\")\n",
+        "print(f\"MSE: {round(MSE, 3)}\")\n",
+        "print(f\"RMSE: {round(RMSE, 3)}\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "mgHTMvw4QlAq"
+      },
+      "source": [
+        "## 4. Verta Set Up"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "S0IjH5jEQCIs"
+      },
+      "outputs": [],
+      "source": [
+        "os.environ['VERTA_HOST'] = ''\n",
+        "os.environ['VERTA_EMAIL'] = ''\n",
+        "os.environ['VERTA_DEV_KEY'] = ''"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "CnIEpTSwQGd1",
+        "outputId": "1e18a6ff-54fd-4a8e-a2f9-8a1863c81629"
+      },
+      "outputs": [],
+      "source": [
+        "client = Client(os.environ['VERTA_HOST'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eNAUDlQmQovE"
+      },
+      "source": [
+        "## 5. Model Register"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Y8Uvjv_bQqhm"
+      },
+      "outputs": [],
+      "source": [
+        "with open('model.pkl', 'wb') as f:\n",
+        "    cloudpickle.dump(gbm, f)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "V9dhdR25QuuQ"
+      },
+      "outputs": [],
+      "source": [
+        "class Predictor(VertaModelBase):\n",
+        "    def __init__(self, artifacts):\n",
+        "        self.model = cloudpickle.load(open(artifacts['serialized_model'], 'rb'))\n",
+        "    \n",
+        "    @verify_io\n",
+        "    def predict(self, data):\n",
+        "        results = []\n",
+        "\n",
+        "        for item in data:\n",
+        "            results.append(self.model.predict(item).tolist())\n",
+        "        \n",
+        "        return results\n",
```

Comment:
Indeed. Done.

Reply:
Could you describe

> otherwise the output in WebApp is not filled

? I think I am able to see API in the WebApp when I use `ModelAPI(X_train, y_train)`, and `y_train` is more accurate than `y_train.to_frame()`.

Reply:
Below are screenshots of the WebApp Input & Output Section for each case.

Using `ModelAPI(X_train, y_train.to_frame(name='PricePrediction'))`:
![Captura de tela de 2023-01-03 11-33-47](https://user-images.githubusercontent.com/2164940/210379364-2ee9a988-57bf-48b1-a9ea-b29f81fa6fb0.png)

Using `ModelAPI(X_train, y_train)`:
![Captura de tela de 2023-01-03 11-36-49](https://user-images.githubusercontent.com/2164940/210379208-5ba90bd8-cc2c-4035-a067-72cb19d2c14d.png)

Notice that the Output section is not filled when `y_train` is not converted to a Dataframe.

How should we proceed?